### PR TITLE
docs(helm): Improve the helm's NOTES.txt

### PR DESCRIPTION
**Pull request by cyriltovena**
**What this PR does / why we need it**:

This PR improves the NOTES.txt shown after installation of the helm chart.

It's a start, but at least covers sending and retrieving logs from the cluster using curl and Grafana.

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [ ] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: production/helm/loki/templates/NOTES.txt [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
  Loki version: {{ .Chart.AppVersion }}
 ***********************************************************************
 
+** Please be patient while the chart is being deployed **
+
+Tip:
+
+  Watch the deployment status using the command: kubectl get pods -w --namespace {{ $.Release.Namespace }}
+
+If pods are too long to schedule make sure pod affinity can be fulfilled in the current cluster.
```
Comment (JStickler) [inline production/helm/loki/templates/NOTES.txt:14]:
```suggestion
If pods are taking too long to schedule make sure pod affinity can be fulfilled in the current cluster.
```

File: production/helm/loki/templates/NOTES.txt [hunk 2]
```diff
@@ -34,3 +46,139 @@ Installed components:
 * query frontend
 {{- end }}
 {{- end }}
+
+
+{{- if eq (include "loki.deployment.isSingleBinary" .) "true" }}
```
Comment (JStickler) [inline production/helm/loki/templates/NOTES.txt:51]: https://github.com/grafana/loki/blob/55b94903415c1bbe3a6cbd7d3a3a5f2d47f4f8a1/production/helm/loki/values.yaml#L30
Should this be `deploymentMode` ?

```suggestion
{{- if eq (include "loki.deploymentMode.isSingleBinary" .) "true" }}
```

File: production/helm/loki/templates/NOTES.txt [hunk 3]
```diff
@@ -34,3 +46,139 @@ Installed components:
...
+
+{{- if .Values.loki.auth_enabled }}
+
+***********************************************************************
+Multi-tenancy
+***********************************************************************
+
+Loki is configured with auth enabled (multi-tenancy) and expects tenant headers (`X-Scope-OrgID`) to be set for all API calls.
+
+Grafana's Loki datasource should be configured using the `HTTP Headers` section with the `X-Scope-OrgID` to target a specific tenant.
```
Comment (JStickler) [inline production/helm/loki/templates/NOTES.txt:170]:
```suggestion
You must configure Grafana's Loki datasource using the `HTTP Headers` section with the `X-Scope-OrgID` to target a specific tenant.
```

File: production/helm/loki/templates/NOTES.txt [hunk 4]
```diff
@@ -34,3 +46,139 @@ Installed components:
...
+For each tenant, you can create a different datasource.
+
+The agent of your choice should also be configured to propagate this header.
```
Comment (JStickler) [inline production/helm/loki/templates/NOTES.txt:173]:
```suggestion
The agent of your choice must also be configured to propagate this header.
```

File: production/helm/loki/templates/NOTES.txt [hunk 5]
```diff
@@ -34,3 +46,139 @@ Installed components:
...
+For example, when using Promtail you can use the `tenant` stage. https://grafana.com/docs/loki/latest/send-data/promtail/stages/tenant/
+
+When not providing the `X-Scope-OrgID` while the auth is enabled, Loki will reject reads and writes with a 404 status code `no org id`.
```
Comment (JStickler) [inline production/helm/loki/templates/NOTES.txt:176]:
```suggestion
When not provided with the `X-Scope-OrgID` while auth is enabled, Loki will reject reads and writes with a 404 status code `no org id`.
```

File: production/helm/loki/templates/NOTES.txt (hunk 2 above)
Comment (cyriltovena) [reply production/helm/loki/templates/NOTES.txt:51]: No this is correct. It's a template. I also tested this with `--dry-run` and all possible variations.

**Review by MichelHollands — APPROVED** [review]
LGTM

**Pull request merged**